src/rdf.rs

```diff
@@ -0,0 +1,277 @@
+use log::{info, warn};
+
+use oxigraph::io::GraphFormat;
+use oxigraph::model::vocab::{rdf, xsd};
+use oxigraph::model::*;
+use oxigraph::store::{QuadIter, SerializerError, Store};
+
+use crate::url::{UrlCheck, UrlType};
+use crate::vocab::{dcat, dcat_mqa, dcterms, dqv};
+
+use crate::error::Error;
+
+// Parse Turtle RDF and load into store
+pub fn parse_turtle(turtle: String) -> Result<Store, Error> {
+    info!("Loading turtle graph");
+
+    let store = Store::new()?;
+    store.load_graph(
+        turtle.as_ref(),
+        GraphFormat::Turtle,
+        GraphNameRef::DefaultGraph,
+        None,
+    )?;
+    Ok(store)
+}
+
+// Retrieve datasets
+pub fn list_datasets(store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        None,
+        Some(rdf::TYPE),
+        Some(dcat::DATASET_CLASS.into()),
+        None,
+    )
+}
+
+// Retrieve distributions of a dataset
+pub fn list_distributions(dataset: NamedNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((dataset).into()),
+        Some((dcat::DISTRIBUTION).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve access urls of a distribution
+pub fn list_access_urls(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcat::ACCESS_URL).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve download urls of a distribution
+pub fn list_download_urls(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcat::DOWNLOAD_URL).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve distribution formats
+pub fn list_formats(distribution: NamedOrBlankNodeRef, store: &Store) -> QuadIter {
+    store.quads_for_pattern(
+        Some((distribution).into()),
+        Some((dcterms::FORMAT).into()),
+        None,
+        None,
+    )
+}
+
+// Retrieve dataset namednode
+pub fn get_dataset_node(store: &Store) -> Option<NamedNode> {
+    list_datasets(&store).next().and_then(|d| match d {
+        Ok(Quad {
+            subject: Subject::NamedNode(n),
+            ..
+        }) => Some(n),
+        _ => None,
+    })
+}
+
+// Map GEO location method
+fn map_format_to_head(format_uri: String) -> String {
+    let fmt = format_uri.split("/").last().unwrap_or_default();
+    match fmt {
+        "WMS_SRVC" => "WMS".to_string(),
+        "WFS_SRVC" => "WFS".to_string(),
+        "WCS_SRVC" => "WCS".to_string(),
+        _ => "HEAD".to_string(),
+    }
```

Comment:
Ikke så farlig, men kan gjøre dette når alle skal gjøres _to string_

```suggestion
        "WMS_SRVC" => "WMS",
        "WFS_SRVC" => "WFS",
        "WCS_SRVC" => "WCS",
        _ => "HEAD",
    }
    .to_string()
```